consensus/src/simplex/actors/voter/mod.rs

```diff
@@ -844,4 +844,246 @@ mod tests {
+                namespace: namespace.clone(),
+                mailbox_size: 128,
+                leader_timeout: Duration::from_millis(500),
+                notarization_timeout: Duration::from_secs(1000),
+                nullify_retry: Duration::from_secs(1000),
+                activity_timeout: 10,
+                replay_buffer: NZUsize!(1024 * 1024),
+                write_buffer: NZUsize!(1024 * 1024),
+                buffer_pool: PoolRef::new(PAGE_SIZE, PAGE_CACHE_SIZE),
+            };
+            let (voter, mut mailbox) = Actor::new(context.clone(), voter_cfg);
+
+            // Resolver and batcher mailboxes
+            let (resolver_sender, _resolver_receiver) = mpsc::channel(8);
+            let resolver_mailbox = resolver::Mailbox::new(resolver_sender);
+            let (batcher_sender, mut batcher_receiver) = mpsc::channel(8);
+            let batcher_mailbox = batcher::Mailbox::new(batcher_sender);
+
+            // Register network channels for the validator
+            let me = participants[0].clone();
+            let (pending_sender, _pending_receiver) = oracle.register(me.clone(), 0).await.unwrap();
+            let (recovered_sender, recovered_receiver) =
+                oracle.register(me.clone(), 1).await.unwrap();
+
+            // Start the actor
+            let handle = voter.start(
+                batcher_mailbox,
+                resolver_mailbox,
+                pending_sender,
+                recovered_sender,
+                recovered_receiver,
+            );
+
+            // Wait for batcher to be notified
+            let message = batcher_receiver.next().await.unwrap();
+            match message {
+                batcher::Message::Update {
+                    current,
+                    leader: _,
+                    finalized,
+                    active,
+                } => {
+                    assert_eq!(current, 1);
+                    assert_eq!(finalized, 0);
+                    active.send(true).unwrap();
+                }
+                _ => panic!("unexpected batcher message"),
+            }
+
+            // Provide almost enough finalize votes
+            let view = 2;
+            let proposal = Proposal::new(
+                Round::new(333, view),
+                view - 1,
+                Sha256::hash(b"finalize_without_notarization"),
+            );
+            let (notarize_votes, expected_notarization) =
+                build_notarization(&schemes, &namespace, &proposal, quorum as usize);
+            let (finalize_votes, expected_finalization) =
+                build_finalization(&schemes, &namespace, &proposal, quorum as usize);
+
+            // Submit just short of enough finalize votes
+            for finalize in finalize_votes.iter().take(quorum as usize - 1).cloned() {
+                mailbox.verified(vec![Voter::Finalize(finalize)]).await;
+            }
+
+            // Submit enough notarize votes to broadcast and force a sync
```

Comment:
While assuming some specific internal behavior is an anti-pattern (specifically syncing before we broadcast a message), I find this to be the "best worst option". It seems just as bad to have the test have access to call sync.

Clean shutdown should handle this sync automatically but we don't have the ability to do that without shutting down the entire context. It would be interesting to drive clean shutdown of just some subset of actors but that is reserved for #1110 .